src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferencesSearchEngine.cs

```diff
@@ -45,6 +51,12 @@ internal partial class FindReferencesSearchEngine
             _options = options;
 
             _progressTracker = progress.ProgressTracker;
+
+            // If we're an explicit invocation, just defer to the threadpool to execute all our work in parallel to get
+            // things done as quickly as possible.  If we're running implicitly, then use a
+            // ConcurrentExclusiveSchedulerPair's exclusive scheduler as that's the most built-in way in the TPL to get
+            // will run things serially.
+            _scheduler = _options.Explicit ? TaskScheduler.Default : new ConcurrentExclusiveSchedulerPair().ExclusiveScheduler;
```

Comment:
💡 It would be good to create a static scheduler for this case, so that two implicit FAR requests (e.g. two CodeLens updates) running concurrently still share a single scheduler.

Reply:
true.  good point.